field/TinyLang/Field/Rename.hs

```diff
@@ -29,10 +37,18 @@ renameVarM var = do
         Just uniq -> setUnique uniq var
 
 withRenamedStatementM :: Statement f -> (Statement f -> RenameM c) -> RenameM c
-withRenamedStatementM (ELet (UniVar uni var) def) cont = do
+withRenamedStatementM (ELet (UniVar uni var) def) kont = do
     defRen <- renameExprM def
-    withFreshenedVar var $ \varFr -> cont $ ELet (UniVar uni varFr) defRen
-withRenamedStatementM (EAssert expr)              cont = renameExprM expr >>= cont . EAssert
+    withFreshenedVar var $ \varFr -> kont $ ELet (UniVar uni varFr) defRen
+withRenamedStatementM (EAssert expr) kont = renameExprM expr >>= kont . EAssert
+withRenamedStatementM (EFor (UniVar uni var) start end stmts) kont = do
+    withFreshenedVar var $ \varFr ->
+        withRenamedStatementsM stmts $ \stmtsRen ->
+            kont $ EFor (UniVar uni varFr) start end stmtsRen
```

Comment:
We need a comment here explaining that we do not have lexical scoping and that is why `kont` is called inside `withRenamedStatementsM stmts`.